src/Microsoft.TestPlatform.CrossPlatEngine/TestEngine.cs

```diff
@@ -71,7 +72,15 @@ public IProxyDiscoveryManager GetDiscoveryManager(IRequestData requestData, ITes
 
             if (this.ShouldRunInNoIsolation(discoveryCriteria.RunSettings, parallelLevel > 1, false))
             {
-                return new InProcessProxyDiscoveryManager(testHostManager);
+                var isTelemetryOptedOut = requestData.IsTelemetryOptedOut;
+                var newRequestData = new RequestData
+                {
+                    MetricsCollection = isTelemetryOptedOut
+                                                     ? (IMetricsCollection)new NoOpMetricsCollection()
```

Comment:
>(IMetricsCollection) [](start = 55, length = 20)

Why is this casting required?

Reply:
For InProcess scenarios creating request Data again.